cmd/subnetcmd/publish.go

```diff
@@ -0,0 +1,380 @@
+
+	if vmDescPath == "" {
+		vm, err = getVMInfo(sc)
+	} else {
+		vm = new(types.VM)
+		err = loadYAMLFile(vmDescPath, vm)
+	}
+	if err != nil {
+		return err
+	}
+
+	// TODO: Publishing exactly 1 subnet and 1 VM in this iteration
+	tsubnet.VMs = []string{vm.Alias}
+
+	subnetYAML, err := yaml.Marshal(tsubnet)
+	if err != nil {
+		return err
+	}
+	vmYAML, err := yaml.Marshal(vm)
+	if err != nil {
+		return err
+	}
+
+	ux.Logger.PrintToUser("Thanks! We got all the bits and pieces. Trying to publish on the provided repo...")
+
+	publisher := publisherCreateFunc(reposDir, repoURL, alias)
+	repo, err := publisher.GetRepo()
+	if err != nil {
+		return err
+	}
+
+	// TODO: if not published? New commit? Etc...
+	if err = publisher.Publish(repo, subnetName, vm.Alias, subnetYAML, vmYAML); err != nil {
+		return err
+	}
+
+	ux.Logger.PrintToUser("Successfully published")
+	return nil
+}
+
+// iterate the reposDir to check what repos already exist locally
+// if nothing is found, prompt the user for an alias for a new repo
+func getAlias(reposDir string) error {
+	// have any aliases already been defined?
+	if alias == "" {
+		matches, err := os.ReadDir(reposDir)
+		if err != nil {
+			return err
+		}
+		if len(matches) == 0 {
+			// no aliases yet; just ask for a new one
+			alias, err = getNewAlias()
+			if err != nil {
+				return err
+			}
+		} else {
+			// there are already aliases, ask how to proceed
+			options := []string{"Provide a new alias", "Pick from list"}
+			choice, err := app.Prompt.CaptureList("Don't know which repo to publish to. How would you like to proceed?", options)
+			if err != nil {
+				return err
+			}
+			if choice == options[0] {
+				// user chose to provide a new alias
+				alias, err = getNewAlias()
+				if err != nil {
+					return err
+				}
+				// double-check: actually this path exists...
+				if _, err := os.Stat(filepath.Join(reposDir, alias)); err == nil {
+					ux.Logger.PrintToUser("The repository with the given alias already exists locally. You may have already published this subnet there (the other explanation is that a different subnet has been published there).")
+					yes, err := app.Prompt.CaptureYesNo("Do you wish to continue?")
+					if err != nil {
+						return err
+					}
+					if !yes {
+						ux.Logger.PrintToUser("User canceled, nothing got published.")
+						return nil
+					}
+				}
+			} else {
+				aliases := make([]string, len(matches))
+				for i, a := range matches {
+					aliases[i] = a.Name()
+				}
+				alias, err = app.Prompt.CaptureList("Pick an alias", aliases)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// ask for a new alias
+func getNewAlias() (string, error) {
+	return app.Prompt.CaptureString("Please provide an alias for the repository we are going to use")
+}
+
+func getRepoURL(reposDir string) error {
+	if repoURL == "" {
+		path := filepath.Join(reposDir, alias)
+		repo, err := git.PlainOpen(path)
+		if err != nil {
```

Comment:
Is not a problem for the tool if there is a directory (ErrRepositoryNotExists not happens) for that repo which has some type of inconsistency and cannot be readed?

Reply:
Ah I see what you mean now. Good question.

There are a series of errors which could return here which are not `ErrRepositoryNotExists`. But the possible errors are of different kind, e.g. filesystem issues or internal git issues. Would have to do a proper scan and discriminate between all possible errors to be able to identify what to do.

I decided therefore to for now just leave a debug log which also logs the error in case this becomes an issue, but to still ignore it for now.